### ISSUE TITLE
Without access on physics options of a Sprite that's a child of PhysicsNode

### ISSUE THREAD

**Issue by lucasglobal** [spritebuilder/SpriteBuilder#777]
![screen shot 2014-08-13 at 0 14 16](https://cloud.githubusercontent.com/assets/3020580/3900870/0b1e1c4c-2298-11e4-96a2-934b21373fbe.png)

As you can see, everything looks good, except for the physics options of my cbfile(sprite) that i have to do some wrong things to activate.

**Comment by LearnCocos2D** [comment]
You are not supposed to activate physics on a CCBFile. The fact that you can do this right now is a bug: https://github.com/spritebuilder/SpriteBuilder/issues/427

Instead open the referenced CCB file and enable physics on the node(s) inside the referenced CCB.

**Comment by lucasglobal** [comment]
Why it's a bug doing this? 

Att,

Lucas Andrade Ribeiro

**Comment by MikeMJ** [comment]
@LearnCocos2D, but activating the physics on the nodes of the CCB has no effect. Physics just doesn't work on that CCB. (As of version 1.2.1) But enabling physics of the the CCB as a whole does work. And you say it's a bug to allow setting physics properties on a CCBFile. If it's disabled totally, how are we going to make physics work?

**Comment by lucasglobal** [comment]
It would be good that when we select that the CCBFile enables physics,it automatically turns itself on a physics node.

**Comment by LearnCocos2D** [comment]
This setup should work:

```
Scene.ccb
-> root node
  -> CCPhysicsNode
    -> CCBFile (references MyNodes.ccb)

MyNodes.ccb
-> root node (physics enabled)
```

Or in MyNodes you can leave the root node physics disabled, but add multiple nodes with physics enabled. For instance:

```
MyNodes.ccb
-> root node (no physics)
  -> chain1 (physics enabled)
  -> chain2 (physics enabled)
  -> chain3 (physics enabled)
  -> chain4 (physics enabled)
```

 Tested with v1.2 and v1.3.

**Comment by LearnCocos2D** [comment]
Here's a visual example, a chain.ccb added to a level which contains the physics node:

![screen shot 2014-10-04 at 19 21 15](https://cloud.githubusercontent.com/assets/214599/4516235/0414a43a-4beb-11e4-8851-616d7eb243bc.png)
(root node does not have physics enabled, all other nodes do)

![screen shot 2014-10-04 at 19 22 02](https://cloud.githubusercontent.com/assets/214599/4516236/07f6ae9a-4beb-11e4-868b-f80fb9445570.png)
(ccphysicsnode has container node (chains & ropes) without physics enabled, containing chains etc which reference chain.ccb but do not have physics enabled themselves)

**Comment by MikeMJ** [comment]
Taking your first solution into consideration, I think there's one thing you're forgetting about. If the type of the CCB file is a CCNode, then there is a root node. But if it's of type CCSprite, then there's no root node and the only thing you can enable physics on is the CCSprite, which as I previously said does not work.

**Comment by LearnCocos2D** [comment]
I can't rule out that there isn't a bug. But I do know that Sprite CCB generally works with physics enabled. Below is my player character (CCSprite root node) which has physics enabled and works fine when dragged into a level CCB as child of a CCPhysicsNode.

![screen shot 2014-10-08 at 10 55 25](https://cloud.githubusercontent.com/assets/214599/4556429/e204760a-4ec8-11e4-805f-40f777c76ea1.png)

The setup is the same as posted earlier:

```
Scene.ccb
-> root node
  -> CCPhysicsNode
    -> CCBFile (references Player.ccb)

Player.ccb
-> root node (physics enabled - see screenshot above)
```

**Comment by MikeMJ** [comment]
Yes, Steffen, you're right. When I tried what you said in another case, it
did work. I can't find any difference between the ones that work, and the
ones that don't. So my guessing is there definitely a bug that needs to be
revised.

**Comment by taycaldwell** [comment]
This bug still seems to exist.

@LearnCocos2D Out of curiosity - For the ball and chain example you gave, if nothing besides the chains and ball of the ccb file has physics enabled, how are you able to hook the chain the the ceiling of the other ccb file? I am trying to implement something similar, and it just falls to the ground. If I wanted to use a pivot node to attach the chain to the ceiling, I can't accomplish this with only physics enabled on chain/ball ccb items.

Edit: Nevermind I now see that having a "hook" on the chain with static physics is the way to go, instead of using a pivot node to attach a chain to a ceiling.

**Comment by LearnCocos2D** [comment]
exactly :)
